ddtrace/span.py

```diff
@@ -165,17 +165,9 @@ def finish(self, finish_time=None):
             self.duration_ns = ft - (self.start_ns or ft)
 
         if self._context:
-            try:
-                self._context.close_span(self)
-            except Exception:
-                log.exception("error recording finished trace")
-            else:
-                # if a tracer is available to process the current context
-                if self.tracer:
-                    try:
-                        self.tracer.record(self._context)
-                    except Exception:
-                        log.exception("error recording finished trace")
+            trace, sampled = self._context.close_span(self)
```

Comment:
Do we no longer need the try/except block?

Reply:
If you look at the codepath, there's really nothing that should be raising